### PR TITLE
Added docs for endpoint commands passing in additional lookup params

### PR DESCRIPTION
**Pull request by TrimBiggs**

### CODE REVIEW

**Comment by TrimBiggs** [comment]
Fixes #558

File: docs/calicoctl/endpoint.md [hunk 1]
```diff
@@ -162,6 +162,11 @@ calicoctl endpoint <ENDPOINT_ID> profile (append|remove|set) [--host=<HOSTNAME>]
     <PROFILES>: One or more profiles to append, remove, or set on the endpoint.
 ```
 
+This command perform the desired action even if the `<ENDPOINT_ID>` is specified
```
Comment (robbrockbank) [inline docs/calicoctl/endpoint.md]: I'd word it so you can use identical text in both cases, so i'd remove the details about what the command does:

"The <ENDPOINT ID> uniquely identifies the endpoint.  Although it is not necessary to include the optional <HOSTNAME>, <ORCHESTRATOR_ID> and <WORKLOAD_ID> identifiers, this command executes faster and with reduced load on the etcd datastore when all of the identifiers are specified together.

Comment (robbrockbank) [inline docs/calicoctl/endpoint.md]: But i'm not too fussed either way.

Wording is a little wrong as it stands though...   "This command perform"   missing s?

File: docs/calicoctl/endpoint.md [hunk 2]
```diff
@@ -162,6 +162,11 @@ calicoctl endpoint <ENDPOINT_ID> profile (append|remove|set) [--host=<HOSTNAME>]
     <PROFILES>: One or more profiles to append, remove, or set on the endpoint.
 ```
 
+This command perform the desired action even if the `<ENDPOINT_ID>` is specified 
+without the `<HOSTNAME>`, `<ORCHESTRATOR_ID>`, or `<WORKLOAD_ID>`.  However, 
+the lookup to retrieve the correct endpoint will execute faster if any or all 
+of these parameters are passed in as well.
```
Comment (robbrockbank) [inline docs/calicoctl/endpoint.md]: I'd just say "all" not "any"

**Pull request merged**